enhancements/network/external-lb-vips.md

```diff
@@ -0,0 +1,213 @@
+    make it re-usable). A new field in `PlatformStatus` will be set.
+5. In machine-config-operator, if LB type is external, the internal loadbalancer will not be
+   deployed.
+6. When using an external loadbalancer, the VIPs fields won't be set in `PlatformStatus` and the
+   OpenShift components won't use it when they're not present and LB type is external.
+7. After the deployment, the IP address of a service (e.g. API) can change, as long as the external
+   loadbalancer is updated and the DNS record has been updated and the results are propagated as well.
+
+### API Extensions
+
+A new field will be added to the `PlatformStatus` of the infrastructure object to tell
+the kind of the loadbalancer.
+
+```golang
+// LoadBalancerType can be either external or internal 
+type LoadBalancerType string
+const (
+	// LoadBalancerTypeExternal
+	LoadBalancerTypeExternal LoadBalancerType = "external"
+	// LoadBalancerTypeInternal
+	LoadBalancerTypeInternal LoadBalancerType = "internal"
+)
+// VSpherePlatformStatus holds the current status of the vSphere infrastructure provider.
+type VSpherePlatformStatus struct {
+  // loadBalancerType defines if a internal or external load balancer
+  // is to be used for the openshift api and ingress endpoints.
+  // When set to Internal the static pods defined in machine config operator
+  // will be used.
+  // When set to External the static pods will not be deployed. It will be 
+  // expected that the LB and DNS A records be pre-configured prior to installation.
+  // +kubebuilder:default=internal
+  // +kubebuilder:validation:Enum=internal;external
+  LoadBalancer LoadBalancerType `json:"loadBalancerType,omitempty"`
+}
+```
+
+This field will be set in the `install-config.yaml`:
+```yaml
+platform:
+  vsphere:
+    loadBalancerType: external
+```
+
+### Implementation Details/Notes/Constraints
+
+* CoreDNS will not be deployed if external loadbalancer is being used.
+  The customer is responsible of populating the DNS records for api, api-int and ingress.
+  We need to check who is responsible for machine records.
+
+* We need to make it ready for any platform who wants to support it, and allow
+  platforms to turn this on in the installer project. This will be achieved while validating the
+  installconfig manifest.
+
+* `PlatformStatus` won't have the VIP fields if LB type is external.
+
+
+### Risks and Mitigations
+
+Minimal risk, some platforms (vsphere, openstack) have been testing it already and no issue
+has been seen so far with this scenario.
+Validations will have to be implemented to make sure that the customer has prepare their
+environment before the IPI deployment.
+
+### Drawbacks
+
+This has been asked by customers, so they can have more flexibility so there is nothing to highlight here.
+
+## Design Details
+
+### Open Questions
+
+* Do we agree on not deploying CoreDNS if an external loadbalancer is being used?
+* Do we want to set the new field in `PlatformSpec` or `PlatformStatus` ?
```

Comment:
I must be missing something. Surely this can only be spec.

Reply:
Like I said in this [comment](https://github.com/openshift/enhancements/pull/1322/files#r1089007230), I'm not sure either and I need Ben / Joel to tell us.

Reply:
As I noted elsewhere, Status is fine for this. The initial implementation should not allow changing this field anyway. It will be set at install time and never changed so there's no need for a Spec field. If/when we support migrating we'll need a spec field, but the MCO templates will still read the value from Status.

Reply:
I left a counter point [here](https://github.com/openshift/enhancements/pull/1322/files#r1090485751), should we converge on a single thread?

If we keep it in status, what would update the status, and would it perform any validation before updating the status field?

Reply:
We agreed that for now it'll be in the `PlatformSpec`.